src/Salt/SiteBundle/Controller/CommentsController.php

```diff
@@ -162,30 +162,70 @@ public function downvoteAction(Comment $comment, UserInterface $user)
      */
     public function exportCommentAction(string $itemType, int $itemId)
     {
-        $url = ($itemType === 'item') ? $this->generateUrl('doc_tree_item_view', ['id' => $itemId], UrlGeneratorInterface::ABSOLUTE_URL) : $this->generateUrl('doc_tree_view', ['slug' => $itemId], UrlGeneratorInterface::ABSOLUTE_URL);
+        $childIds = [];
         $repo = $this->getDoctrine()->getManager()->getRepository(Comment::class);
-        $headers = ['Framework Name', 'Node Address', ($itemType === 'item') ? 'HumanCodingScheme' : null, 'User', 'Organization', 'Comment', 'Created Date', 'Updated Date'];
-        $rows[] = implode(',', array_filter($headers));
-        $comment_data = $repo->findBy([$itemType => $itemId]);
+        $headers = ['Framework Name', 'Node Address', 'HumanCodingScheme', 'User', 'Organization', 'Comment', 'Created Date', 'Updated Date'];
+        $rows[] = implode(',', $headers);
+        $lsItemRepo = $this->getDoctrine()->getManager()->getRepository(LsItem::class);
+        switch ($itemType)
+        {
+            case 'document':
+            {
+                $comment_data = $repo->findBy([$itemType => $itemId]);
+                $child_comment_rows = $this->csvArray($comment_data, $itemType);
+                foreach ($child_comment_rows as $child_row)
+                {
+                    $rows[] = $child_row;
+                }
+                $lsDoc = $this->getDoctrine()->getManager()->getRepository(LsDoc::class)->findOneById($itemId);
+                $lsDocChilds = $lsItemRepo->findBylsDoc($lsDoc->getId());
+                if (!empty($lsDocChilds)){
+                    foreach ($lsDocChilds as $lsDocChild){
+                        $childIds[] = $lsDocChild->getId();
+                    }
+                }
+                array_filter($childIds);
+                break;
+            }
+            case 'item':
+            {
+                $lsItem = $lsItemRepo->findOneById($itemId);
+                $lsItemRepo->allChilds($childIds, $lsItem);
+                $childIds[] = $itemId;
+                 break;
+            }
+        }
+        $comment_data = $repo->findBy(['item' => $childIds]);
+        $child_comment_rows = $this->csvArray($comment_data, 'item');
+        if (!empty($child_comment_rows))
+        {
+            foreach ($child_comment_rows as $child_row)
+            {
+                $rows[] = $child_row;
+            }
+        }
+        $content = implode("\n", $rows);
+        $response = new Response($content);
+        $response->headers->set('content_type', 'text/csv');
```

Comment:
Might want to add a Content-Disposition to have the result automatically saved as an attachment instead of being displayed in the browser.